cluster/src/main/java/org/apache/iotdb/cluster/qp/executor/QueryMetadataExecutor.java

```diff
@@ -177,6 +185,123 @@ public String processMetadataInStringQuery()
     return combineMetadataInStringList(metadataList);
   }
 
+  public Metadata processMetadataQuery()
+      throws InterruptedException, ProcessorException, PathErrorException {
+    Set<String> groupIdSet = router.getAllGroupId();
+
+    Metadata[] metadatas = new Metadata[groupIdSet.size()];
+    List<SingleQPTask> taskList = new ArrayList<>();
+    for (String groupId : groupIdSet) {
+      QueryMetadataRequest request = new QueryMetadataRequest(groupId,
+          readMetadataConsistencyLevel);
+      SingleQPTask task = new SingleQPTask(false, request);
+      taskList.add(task);
+
+      LOGGER.debug("Execute query metadata statement for group {}.", groupId);
+      /** Check if the plan can be executed locally. **/
+      if (canHandleQueryByGroupId(groupId)) {
+        LOGGER.debug("Execute query metadata statement locally for group {}.", groupId);
+        asyncQueryMetadataLocally(groupId, task);
+      } else {
+        try {
+          PeerId holder = RaftUtils.getRandomPeerID(groupId);
+          asyncSendNonQueryTask(task, holder, 0);
+        } catch (RaftConnectionException e) {
+          LOGGER.error(e.getMessage());
+          throw new ProcessorException("Raft connection occurs error.", e);
+        }
+      }
+    }
+    for (int i = 0; i < taskList.size(); i++) {
+      SingleQPTask task = taskList.get(i);
```

Comment:
So you submit these tasks serially? Second task won't be submitted until first task is done?

Reply:
No, the submission process has already been done before this for loop, here we just get results from taskList.